src/nplinker/logconfig.py

```diff
@@ -14,20 +14,21 @@
 
 import logging
 import sys
+from typing_extensions import Self
 
 
 class LogConfig():
 
-    active_loggers = {}
+    active_loggers: dict[Self, logging.Logger] = {}
     logfmt = '%(asctime)s [%(levelname)s] %(filename)s:%(lineno)d, %(message)s'
     default_loglevel = logging.INFO
     # default destination for new Loggers
     default_logdest = logging.StreamHandler(sys.stdout)
     # additional destinations to be added to new Loggers
-    additional_logdests = []
+    additional_logdests: list[logging.Logger] = []
 
     @staticmethod
-    def getLogger(obj, level=default_loglevel, dest=default_logdest):
+    def getLogger(obj: Self, level=default_loglevel, dest=default_logdest) -> logging.Logger:
```

Comment:
line 40 `logger = logging.Logger(obj)` is misusing Logger class, leading to problems of catching log output. 
I just changed it to `logging.getLogger(obj)` method in my local code, and so the `obj` here should be `str`.

Reply:
See https://github.com/NPLinker/nplinker/pull/91/commits/190c433171355a1bd7607caba32397135292d7eb